resources/assets/js/components/Application/Experience/Experience.tsx

```diff
@@ -0,0 +1,742 @@
+    setDisconnectedRequiredSkills(
+      getDisconnectedRequiredSkills(
+        experiences,
+        experienceSkills,
+        essentialSkills,
+      ),
+    );
+    setExperienceData(null);
+    setIsModalVisible({ id: "", visible: false });
+  };
+
+  const submitExperience = (data: ExperienceSubmitData): Promise<void> =>
+    handleSubmitExperience(data).then(closeModal);
+
+  const editExperience = (
+    id: string,
+    experience: Experience,
+    savedOptionalSkills: Skill[],
+    savedRequiredSkills: Skill[],
+  ): void => {
+    setExperienceData({
+      ...experience,
+      savedOptionalSkills,
+      savedRequiredSkills,
+    });
+    setIsModalVisible({ id, visible: true });
+  };
+
+  const deleteExperience = (
+    id: number,
+    type: Experience["type"],
+  ): Promise<void> => handleDeleteExperience(id, type).then(closeModal);
+
+  const softSkills = [...assetSkills, ...essentialSkills].filter(
+    (skill) => skill.skill_type_id === SkillTypeId.Soft,
+  );
+
+  const modalButtons = {
+    education: {
+      id: "experience_education",
+      title: intl.formatMessage(educationMessages.modalTitle),
+      icon: "fas fa-book",
+    },
+    work: {
+      id: "experience_work",
+      title: intl.formatMessage(workMessages.modalTitle),
+      icon: "fas fa-briefcase",
+    },
+    community: {
+      id: "experience_community",
+      title: intl.formatMessage(communityMessages.modalTitle),
+      icon: "fas fa-carry",
+    },
+    personal: {
+      id: "experience_personal",
+      title: intl.formatMessage(personalMessages.modalTitle),
+      icon: "fas fa-mountain",
+    },
+    award: {
+      id: "experience_award",
+      title: intl.formatMessage(awardMessages.modalTitle),
+      icon: "fas fa-trophy",
+    },
+  };
+
+  const modalButton = ({ id, title, icon }): React.ReactElement => (
+    <div key={id} data-c-grid-item="base(1of2) tp(1of3) tl(1of5)">
+      <button
+        className="application-experience-trigger"
+        data-c-card
+        data-c-background="c1(100)"
+        data-c-radius="rounded"
+        title={title}
+        data-c-dialog-id={id}
+        data-c-dialog-action="open"
+        type="button"
+        onClick={(): void => openModal(id)}
+      >
+        <i className={icon} aria-hidden="true" />
+        <span data-c-font-size="regular" data-c-font-weight="bold">
+          {title}
+        </span>
+      </button>
+    </div>
+  );
+
+  const modalRoot = document.getElementById("modal-root");
+
+  const experienceAccordion = (
+    experienceType: string,
+    experience: Experience,
+    irrelevantSkillCount: number,
+    relevantSkills: RelevantSkill[],
+    handleEdit: () => void,
+    handleDelete: () => void,
+  ): React.ReactElement => {
+    const education = experience as ExperienceEducation;
+    const educationType =
+      educationTypes.find(({ id }) => education.education_type_id === id)?.name[
+        locale
+      ] || "Education type not found";
```

Comment:
Need to localize these error messages, as well as the general `<p>Error</p>` tag further down